### ISSUE TITLE
Feature: Try a model prediction uncertainty coefficient

### ISSUE THREAD

**Issue by luciaquirke** [EleutherAI/bergson#43]
As in the continual learning baselines in https://www.trentonbricken.com/Continual-Learning-Baselines/

data attribution methods need to learn which weights are important, and work on training tokens that have been well-learned (gradients near zero). 

> A number of influential continual learning algorithms like [Elastic Weight Consolidation](https://arxiv.org/abs/1612.00796) (EWC) and [Synaptic Intelligence](https://arxiv.org/abs/1703.04200?context=cs) (SI) protect neural network weights important for previous tasks from being updated by newer tasks. Otherwise, these network weights are overwritten by the next task, resulting in catastrophic forgetting.
> It turns out that existing implementations of these algorithms have, at least for some tasks, been significantly underestimating their performance. This is because they need a small modification to be able to actually learn which network weights need protecting. In a Split MNIST class incremental benchmark, this modification leads to 43% higher validation accuracy.
> These weight regularization methods use the magnitude of gradients during the backwards pass to infer what weights are important for a particular task. They then use a regularization term in the loss function to penalize the model from updating these weights during new tasks. However, in cases where the model performs very well on all training data within a task, there is almost no gradient for the model to be able to learn what weights are important!
> In order to restore gradient information, we introduce a β coefficient into the cross-entropy loss function when learning weight importances. This is used to make the model less confident in its predictions. A hyperparameter sweep found that β=0.005 worked the best for both EWC and SI, leading to 43% and 15% performance gains as shown in the below figure.

A variation on this experiment is to use the top-k least frequent or highest entropy tokens in the sequence to calculate the sequence gradient.
